<a href="https://colab.research.google.com/github/pedrojrcruz30-code/ada---projeto---modulo-2/blob/Pedro/projeto_pedro.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [13]:
def salvar_banco_csv(df, caminho):
    # sua solução aqui
    df.to_csv(caminho, index=False)
    return True


def carregar_banco_csv(caminho):
    # sua solução aqui
    return pd.read_csv(caminho)


# Projeto grupo 2


# Contexto e Metodologia
Análise baseada em dados reais de 2024 focada em cargos de tecnologia, níveis de experiência e modelos de trabalho.

## Variáveis de Perfil
Cargos (Job Title), Nível de Experiência e Localização (Company Location) compõem a base do perfil profissional.

##Métricas Salariais
Análise padronizada em USD para comparação global, cobrindo salários anuais de 100k a 150k+.

In [15]:
import pandas as pd
import numpy as np

df = carregar_banco_csv('data_science_salaries.csv')

df.head()

,job_title,experience_level,employment_type,work_models,work_year,employee_residence,salary,salary_currency,salary_in_usd,company_location,company_size
0,Data Engineer,Mid-level,Full-time,Remote,2024,United States,148100,USD,148100,United States,Medium
1,Data Engineer,Mid-level,Full-time,Remote,2024,United States,98700,USD,98700,United States,Medium
2,Data Scientist,Senior-level,Full-time,Remote,2024,United States,140032,USD,140032,United States,Medium
3,Data Scientist,Senior-level,Full-time,Remote,2024,United States,100022,USD,100022,United States,Medium
4,BI Developer,Mid-level,Full-time,On-site,2024,United States,120000,USD,120000,United States,Medium


# Consultando a tabela padrão

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6599 entries, 0 to 6598
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   job_title           6599 non-null   object
 1   experience_level    6599 non-null   object
 2   employment_type     6599 non-null   object
 3   work_models         6599 non-null   object
 4   work_year           6599 non-null   int64 
 5   employee_residence  6599 non-null   object
 6   salary              6599 non-null   int64 
 7   salary_currency     6599 non-null   object
 8   salary_in_usd       6599 non-null   int64 
 9   company_location    6599 non-null   object
 10  company_size        6599 non-null   object
dtypes: int64(3), object(8)
memory usage: 567.2+ KB


# 1 - Fazendo alguns ajustes na tabela

### Excluindo algumas colunas e visualizando as 15 primeiras linhas

In [17]:
nova_df = df.drop(columns=['employment_type', "salary_currency", "experience_level"])

nova_df.head(15)


,job_title,work_models,work_year,employee_residence,salary,salary_in_usd,company_location,company_size
0,Data Engineer,Remote,2024,United States,148100,148100,United States,Medium
1,Data Engineer,Remote,2024,United States,98700,98700,United States,Medium
2,Data Scientist,Remote,2024,United States,140032,140032,United States,Medium
3,Data Scientist,Remote,2024,United States,100022,100022,United States,Medium
4,BI Developer,On-site,2024,United States,120000,120000,United States,Medium
5,BI Developer,On-site,2024,United States,62100,62100,United States,Medium
6,Research Analyst,On-site,2024,United States,250000,250000,United States,Medium
7,Research Analyst,On-site,2024,United States,150000,150000,United States,Medium
8,Data Engineer,Remote,2024,United States,219650,219650,United States,Medium
9,Data Engineer,Remote,2024,United States,136000,136000,United States,Medium


### Consultando as informações da nova tabela


In [ ]:
nova_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6599 entries, 0 to 6598
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   job_title           6599 non-null   object
 1   work_models         6599 non-null   object
 2   work_year           6599 non-null   int64 
 3   employee_residence  6599 non-null   object
 4   salary              6599 non-null   int64 
 5   salary_in_usd       6599 non-null   int64 
 6   company_location    6599 non-null   object
 7   company_size        6599 non-null   object
dtypes: int64(3), object(5)
memory usage: 412.6+ KB


### Organizando as colunas em lista

In [19]:
display(nova_df.columns)

print("="*150)

list(nova_df.columns)



Index(['job_title', 'work_models', 'work_year', 'employee_residence', 'salary',
       'salary_in_usd', 'company_location', 'company_size'],
      dtype='object')

['job_title',
 'work_models',
 'work_year',
 'employee_residence',
 'salary',
 'salary_in_usd',
 'company_location',
 'company_size']

### COUNT – QUANTIDADE DE VALORES
### MEAN – MEDIA DE VALORES
### STD –  DESVIO PADRAO
### MIN – VALOR MINIMO
### 25% -
### 50% -
### 75 %-
### MAX – VALOR MAXIMO



In [20]:
nova_df.describe()

,work_year,salary,salary_in_usd
count,6599.000000,6.599000e+03,6599.000000
mean,2022.818457,1.792833e+05,145560.558569
std,0.674809,5.263722e+05,70946.838070
min,2020.000000,1.400000e+04,15000.000000
25%,2023.000000,9.600000e+04,95000.000000
50%,2023.000000,1.400000e+05,138666.000000
75%,2023.000000,1.875000e+05,185000.000000
max,2024.000000,3.040000e+07,750000.000000


### Quantidade de funcionarios por cargo

In [21]:
nova_df.job_title.value_counts()

,count
job_title,
Data Engineer,1307
Data Scientist,1243
Data Analyst,910
Machine Learning Engineer,629
Analytics Engineer,246
...,...
Data Science Tech Lead,1
Staff Machine Learning Engineer,1
Cloud Data Architect,1


## Apos analise identificamos que de todos os funcionarios da ativa (6598), 3813 trabalham presncialmente, 2561 remotamente e no regime hibrido 225.



In [23]:
modelo_trabalho = df['work_models'].value_counts()

display(modelo_trabalho)

modelo_trabalho = (df["work_models"].value_counts(normalize=True).map("{:.1%}".format))

display(modelo_trabalho)

total_funcionarios = df['job_title'].count()

display(total_funcionarios)



,count
work_models,
On-site,3813
Remote,2561
Hybrid,225


,proportion
work_models,
On-site,57.8%
Remote,38.8%
Hybrid,3.4%


np.int64(6599)

# As 5 maiores medias salariais

In [24]:
df_media_cargo = df.groupby('job_title')['salary_in_usd'].mean().round(2)\
                .reset_index().sort_values(by="salary_in_usd", ascending=False)

display(df_media_cargo)


,job_title,salary_in_usd
9,Analytics Engineering Manager,399880.00
72,Data Science Tech Lead,375000.00
111,Managing Director Data Science,280000.00
7,AWS Data Architect,258000.00
29,Cloud Data Architect,250000.00
...,...,...
109,Machine Learning Specialist,55000.00
92,Insight Analyst,46339.83
32,Compliance Data Analyst,45000.00
118,Principal Data Architect,38154.00


# As 5 menores medias salarias


In [25]:
df_media_cargo.tail()

,job_title,salary_in_usd
109,Machine Learning Specialist,55000.00
92,Insight Analyst,46339.83
32,Compliance Data Analyst,45000.00
118,Principal Data Architect,38154.00
37,Data Analyst Lead,18000.00


# 1. Progressão salarial por nível de experiência

In [27]:
# --- Parte A: Salário médio por nível de experiência geral ---
print("--- Média Salarial por Nível de Experiência Geral ---")
# Agrupa por nível de experiência e calcula a média do salário em USD
media_por_experiencia = df.groupby('experience_level')['salary_in_usd'].mean().round(2).reset_index()
display(media_por_experiencia)

print("\n" + "="*50 + "\n")

# --- Parte B: Comparação entre os cargos principais ---
print("--- Progressão Salarial para os Principais Cargos ---")

# 1. Definimos a lista dos cargos que queremos filtrar
cargos_principais = ['Data Analyst', 'Data Scientist', 'Data Engineer']

# 2. Filtramos o DataFrame original para conter apenas esses cargos
df_filtrado_cargos = df[df['job_title'].isin(cargos_principais)]

# 3. Agrupamos por cargo E nível de experiência para ver a progressão de cada um
progressao_cargos = df_filtrado_cargos.groupby(['job_title', 'experience_level'])['salary_in_usd'].mean().unstack().round(2)

# Exibe o resultado em formato de tabela cruzada para facilitar a comparação visual
display(progressao_cargos)

--- Média Salarial por Nível de Experiência Geral ---


,experience_level,salary_in_usd
0,Entry-level,84448.92
1,Executive-level,189687.35
2,Mid-level,119019.81
3,Senior-level,162071.06




--- Progressão Salarial para os Principais Cargos ---


experience_level,Entry-level,Executive-level,Mid-level,Senior-level
job_title,,,,
Data Analyst,78290.84,114107.14,94854.81,122400.54
Data Engineer,79441.09,195357.38,117253.14,156129.57
Data Scientist,83387.10,189050.52,115501.05,169793.45


# 2. Trabalho remoto compensa?

In [28]:
# Parte A: Comparação Geral por Modelo de Trabalho ---
print("--- Média Salarial por Modelo de Trabalho ---")

media_por_modelo = df.groupby('work_models')['salary_in_usd'].mean().round(2).reset_index()
display(media_por_modelo)

display("="*150)

# Parte B: Modelo de Trabalho Cruzado com Nível de Experiência ---
print("--- Média Salarial por Modelo de Trabalho e Nível de Experiência ---")

# Agrupamos por ambas as colunas e calculamos a média
modelo_e_experiencia = df.groupby(['experience_level', 'work_models'])['salary_in_usd'].mean().unstack().round(2)
display(modelo_e_experiencia)

--- Média Salarial por Modelo de Trabalho ---


,work_models,salary_in_usd
0,Hybrid,83293.50
1,On-site,152600.10
2,Remote,140550.14


'======================================================================================================================================================'

--- Média Salarial por Modelo de Trabalho e Nível de Experiência ---


work_models,Hybrid,On-site,Remote
experience_level,,,
Entry-level,63998.44,96933.89,75154.08
Executive-level,147368.67,185061.71,198580.90
Mid-level,76990.86,129770.81,107083.53
Senior-level,106145.92,166657.46,157343.06


#3. **Onde estão as melhores oportunidades?**
###Quais países da residência do empregado têm os maiores salários médios? Existe correlação entre país e modalidade de trabalho (remoto vs presencial)?

In [30]:
# Agrupando por país de residência e calculando a média salarial em USD
maiores_salarios = df.groupby('employee_residence')['salary_in_usd'].mean().round(2).sort_values(ascending=False)

# Exibindo os 5 principais países
print("Países com os maiores salários médios (em USD):")
display(maiores_salarios.head())

Países com os maiores salários médios (em USD):


,salary_in_usd
employee_residence,
Israel,417937.00
Qatar,300000.00
Malaysia,200000.00
Puerto Rico,166000.00
United States,157780.94


**Sugestão de commit:**

```
feat: adiciona suporte a CSV para entrada e saida de dados

Permite persistir o banco em disco e carregar de volta. Em
qualquer cenario real de analise, os dados nao vem como
dicionarios Python: vem de arquivos. Esse e o primeiro passo
para a aula 5, onde o sistema vai operar sobre um dataset real
do mercado de dados.
```